src/Compilers/CSharp/Portable/Errors/MessageID.cs

```diff
@@ -204,6 +205,7 @@ internal static LanguageVersion RequiredVersion(this MessageID feature)
             {
                 // C# 8 features.
                 case MessageID.IDS_FeatureStaticNullChecking: // syntax and semantic check
+                case MessageID.IDS_FeatureObjectGenericTypeConstraint:   // semantic check
```

Comment:
>IDS_FeatureObjectGenericTypeConstraint [](start = 31, length = 38)

Why use a separate feature? I expected this would be part of the nullable reference types feature.

Reply:
> Why use a separate feature? I expected this would be part of the nullable reference types feature.

I think it is better to keep this as a separate feature, like other new constraints that syntactically aren't tied to nullable feature (don't use ```?```). For example, I believe an experience is better when one is able to use this constraint without any diagnostics regardless of NonNullTypes context. And having different error for the case when feature is not enabled feels better too.

---
In reply to: [217125534](https://github.com/dotnet/roslyn/pull/29809#discussion_r217125534) [](ancestors = 217125534)